### PR TITLE
Initial trials for ThemeProvider

### PR DESCRIPTION
**Pull request by iRoachie**
Right now this is proof of concept for a ThemeProvider. Right now it only accepts overrides for the base colours. 

In the future, we'll need to add other fields as well as better naming. e.g(checkboxColor, badgeBackgroundColor)

#### Pending
- [ ] Example

### CODE REVIEW

**Comment by iRoachie** [comment]
Updated description for better gist.

File: src/config/getTheme.js [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    checkbox: StyleSheet.create({
+      wrapper: {
+        flexDirection: 'row',
+        alignItems: 'center',
+      },
+      container: {
+        margin: 5,
+        marginLeft: 10,
+        marginRight: 10,
+        backgroundColor: '#fafafa',
```
Comment (binoy14) [inline src/config/getTheme.js]: Should this be using the colors object?

Comment (iRoachie) [inline src/config/getTheme.js]: Not sure, it wasn’t using it before

Comment (binoy14) [inline src/config/getTheme.js]: Not sure what it affects but probably a good idea to take a look what it's doing

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "editor.formatOnSave": true,
+  "prettier.eslintIntegration": true
+}
```
Comment (Monte9) [inline .vscode/settings.json]: this should not be committed. Please add this file to `.gitignore`

Comment (iRoachie) [reply .vscode/settings.json]: Ooops

**Comment by iRoachie** [comment]
Taking a step back and going to make an easier way to show how it works, and an upgrade path for making it available for all components

**Comment by Monte9** [comment]
Ok, that's good to know @iRoachie! Let's plan to swing back to this after we do the v1 release. That way we will already have the new API changes & components out and being used. 

So this theme PR will be easier to implement & release.

**Comment by iRoachie** [comment]
Sounds good to me!

**Pull request closed without merging**